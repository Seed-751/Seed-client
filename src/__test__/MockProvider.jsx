import React from "react";
import PropTypes from "prop-types";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import GrobalStyles from "../styles/index";
import theme from "../styles/theme";
import store from "../reducers/store";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

export default function MockProvider({ children }) {
  const history = createMemoryHistory();

  return (
    <Provider store={store}>
      <Router history={history}>
        <ThemeProvider theme={theme}>
          <GrobalStyles />
          {children}
        </ThemeProvider>
      </Router>
    </Provider>
  );
}

MockProvider.propTypes = {
  children: PropTypes.element.isRequired,
};
