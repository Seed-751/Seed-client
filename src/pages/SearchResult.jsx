import React, { useState } from "react";
import { useSelector } from "react-redux";
import styled from "styled-components";

import MusicList from "../components/MusicList";
import NoResult from "../components/NoResult";
import { selectSearch } from "../reducers/searchSlice";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  padding: 10px;
  gap: 30px;

  .title {
    font-size: 50px;
  }
`;

const Nav = styled.nav`
  ul {
    display: flex;
    flex-direction: row;
    font-size: 30px;
    gap: 20px;
  }
`;

const LinkButton = styled.button`
  color: ${({ active, theme }) => active ? "white" : theme.color.lightGray};
  font-size: 30px;
  border-bottom: ${({ active }) => active ? "2px solid white" : "none"};
`;

export default function SearchResult() {
  const [active, setActive] = useState({
    byTitle: true,
  });
  const { searchResult } = useSelector(selectSearch);
  const { albumsByTitle, albumsByArtist } = searchResult;

  function handleShowByTitle() {
    setActive({ byTitle: true });
  }

  function handleShowByArtist() {
    setActive({ byArtist: true });
  }

  return (
    <Container>
      <Nav>
        <ul>
          <LinkButton active={active.byTitle} onClick={handleShowByTitle}>Title</LinkButton>
          <LinkButton active={active.byArtist} onClick={handleShowByArtist}>Artist</LinkButton>
        </ul>
      </Nav>
      {active.byTitle &&
        <>
          {albumsByTitle?.length
            ? <MusicList data={albumsByTitle} />
            : <NoResult />
          }
        </>
      }
      {active.byArtist &&
        <>
          {albumsByArtist?.length
            ? <MusicList data={albumsByArtist} />
            : <NoResult />
          }
        </>
      }
    </Container>
  );
}
