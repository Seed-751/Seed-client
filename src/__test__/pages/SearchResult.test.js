import React from "react";
import { render } from "@testing-library/react";

import SearchResult from "../../pages/SearchResult";
import MockProvider from "../MockProvider";

describe("Search Result page", () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("matches snapshot", () => {
    const { container } = render(
      <MockProvider>
        <SearchResult />
      </MockProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
