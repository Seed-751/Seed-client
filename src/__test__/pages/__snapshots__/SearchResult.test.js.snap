// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Result page matches snapshot 1`] = `
<div>
  <div
    class="sc-jrAFXE jynXyz"
  >
    <nav
      class="sc-kEjbQP eVNQAr"
    >
      <ul>
        <button
          class="sc-iqHYmW gwycnj"
        >
          Title
        </button>
        <button
          class="sc-iqHYmW fGfoqI"
        >
          Artist
        </button>
      </ul>
    </nav>
    <div
      class="sc-pGacB kGKRFi"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 12h-2v-2h2v2zm0-4h-2V6h2v4z"
        />
      </svg>
      <strong>
        검색 결과를 찾지 못했습니다
      </strong>
      <p>
        다시 시도해주세요
      </p>
    </div>
  </div>
</div>
`;
